src/Sign.Cli/CodeCommand.cs

```diff
@@ -114,5 +115,20 @@ private static HashAlgorithmName ParseHashAlgorithmName(ArgumentResult result)
                     return HashAlgorithmName.SHA256;
             }
         }
+
+        private static Uri? ParseUrl(ArgumentResult result)
+        {
+            if (result.Tokens.Count != 1 ||
+                !Uri.TryCreate(result.Tokens[0].Value, UriKind.Absolute, out Uri? uri) ||
+                !(string.Equals(Uri.UriSchemeHttp, uri.Scheme, StringComparison.OrdinalIgnoreCase) ||
+                    string.Equals(Uri.UriSchemeHttps, uri.Scheme, StringComparison.OrdinalIgnoreCase)))
+            {
+                result.ErrorMessage = $"Unsupported value for --{result.Argument.Name}.  The value must be an HTTP or HTTPS URL.";
```

Comment:
The error message doesn't seem to match the validations above. There are checks that
* Check number of tokens (instances of an argument, I presume?);
* Check if the URL is valid and is absolute;
* Check if the URL schema is HTTP or HTTPS.
The error message only mentions the last requirement of those 3.

Reply:
It would be especially confusing to see this message when there is more than one url artument.

Reply:
I added "absolute" to the error message (your # 2).

> It would be especially confusing to see this message when there is more than one url artument.

You won't.  System.CommandLine will report `Option '--timestamp-url' expects a single argument but 2 were provided.`